ocp_resources/ip_address_pool.py

```diff
@@ -0,0 +1,54 @@
+# API reference: https://metallb.universe.tf/apis/#metallb.io/v1beta1.IPAddressPool
+
+from ocp_resources.resource import NamespacedResource
+
+
+class IPAddressPool(NamespacedResource):
+    """
+    IPAddressPool object.
+    """
+
+    api_group = NamespacedResource.ApiGroup.METALLB_IO
+
+    def __init__(
+        self,
+        name=None,
+        namespace=None,
+        client=None,
+        addresses=None,
+        auto_assign=True,
+        avoid_buggy_ips=False,
+        **kwargs,
+    ):
+        """
+        Args:
+            name (str): Name of the MetalLB or it's CR's.
+            namespace (str): Namespace of the MetalLB
+            client: (DynamicClient): DynamicClient to use.
+            addresses (list): List of IP addresses.
+            auto_assign (bool): IP address assignment. (default: True)
+            avoid_buggy_ips (bool): Avoid buggy IP address (default: False)
+        """
+        super().__init__(
+            name=name,
+            namespace=namespace,
+            client=client,
+            **kwargs,
+        )
+        if not addresses:
```

Comment:
Sorry, my bad
you need to check this only if not yaml_file, so after line 47

Reply:
Np! I will update it.

```
    def to_dict(self):
        super().to_dict()
        if not self.yaml_file:
            if not self.addresses:
                raise ValueError("'addresses' cannot be None")
            self.res["spec"] = {"addresses": self.addresses}
```

Reply:
Updated it.